### PR TITLE
COM-363: Update Starter@next to latest canary release

### PR DESCRIPTION
**Pull request by fraxachun**
- Implement UserPermissions-module
- Use updated SitesConfig-type

(Replacement for https://github.com/vivid-planet/comet-starter/pull/127)

### CODE REVIEW

File: admin/src/config.ts [hunk 1]
```diff
@@ -25,4 +27,6 @@ export function createConfig() {
     };
 }
 
+export type SitesConfig = Record<string, SiteConfig>;
```
Comment (johnnyomair) [inline admin/src/config.ts:30]: You could add a type cast to `JSON.parse(environmentVariables.SITES_CONFIG)`, which would allow SitesConfigProvider to infer the type

File: api/src/app.module.ts [hunk 2]
```diff
@@ -67,12 +66,20 @@ export class AppModule {
...
-                        return user.domains.includes(requestScope.domain);
-                    },
+                AuthModule.forRoot(config),
+                UserPermissionsModule.forRootAsync({
+                    useFactory: (userService: UserService, accessControlService: AccessControlService) => ({
+                        availablePermissions: ["products"],
+                        availableContentScopes: [
+                            { domain: "main", language: "de" },
+                            { domain: "main", language: "en" },
+                            { domain: "secondary", language: "en" },
```
Comment (johnnyomair) [inline api/src/app.module.ts:76]:
```suggestion
                            { domain: "secondary", language: "de" },
                            { domain: "secondary", language: "en" },
```

File: api/src/app.module.ts [hunk 3]
```diff
@@ -67,12 +66,20 @@ export class AppModule {
...
+                        ],
+                        userService,
+                        accessControlService,
+                    }),
+                    inject: [UserService, AccessControlService],
+                    imports: [AuthModule.forRoot(config)],
```
Comment (johnnyomair) [inline api/src/app.module.ts:82]: Doesn't this create two auth modules, and therefore two instances of all services?

File: api/src/auth/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { FindUsersArgs, User, UserPermissionsUserServiceInterface, Users } from "@comet/cms-api";
+import { Injectable } from "@nestjs/common";
+
+import { staticUsers } from "./static-users";
+
+@Injectable()
+export class UserService implements UserPermissionsUserServiceInterface {
```
Comment (johnnyomair) [inline api/src/auth/user.service.ts:7]: Should we add a note here that this needs to be adapted in the application?

File: create-app/package.json [hunk 5]
```diff
@@ -37,7 +37,7 @@
         "rimraf": "^5.0.1"
     },
     "devDependencies": {
-        "@comet/eslint-config": "6.0.0-canary-20231122070819",
+        "@comet/eslint-config": "6.0.0-canary-20240126133306",
```
Comment (johnnyomair) [inline create-app/package.json:40]: You need to run install in this directory please

File: package-lock.json [hunk 6]
```diff
@@ -17,34 +17,35 @@
             }
         },
         "node_modules/@babel/code-frame": {
-            "version": "7.18.6",
```
Comment (johnnyomair) [inline package-lock.json:20]: Why do you have changes in the root package-lock.json, but not in package.json?

File: api/src/auth/user.service.ts (hunk 4 above)
Comment (nsams) [reply api/src/auth/user.service.ts:7]: This service needs to be named UserLocalService or so and only be used by AuthLocal.

And what default user service for non-local could the starter have?

File: package-lock.json (hunk 6 above)
Comment (johnnyomair) [reply package-lock.json:20]: It's not necessary for updating packages, but it doesn't hurt anyway. Would only be a problem if you're accidentally using the wrong NPM version.

File: api/src/auth/user.service.ts (hunk 4 above)
Comment (johnnyomair) [reply api/src/auth/user.service.ts:7]: > And what default user service for non-local could the starter have?

We could implement a basic user management module in the Starter/Demo. Probably will be deleted in most of our projects, but could be part of the showcase. Of course, out of scope of this PR.

Comment (johnnyomair) [reply api/src/auth/user.service.ts:7]: We'd need a guide about how to make the Starter production-ready then. We should discuss this in tomorrow's meeting.

**Review by johnnyomair — APPROVED** [review]

File: api/src/auth/permission.interface.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+declare module "@comet/cms-api" {
+    interface Permission {
+        products: string;
+    }
+}
+
+export {};
```
Comment (dkarnutsch) [inline api/src/auth/permission.interface.ts:7]: why that?

File: site/package.json [hunk 8]
```diff
@@ -26,11 +26,11 @@
     },
     "dependencies": {
         "@comet/cms-site": "6.0.0-canary-20231122070819",
-        "@opentelemetry/api": "^1.4.0",
-        "@opentelemetry/auto-instrumentations-node": "^0.36.1",
-        "@opentelemetry/exporter-trace-otlp-http": "^0.35.1",
-        "@opentelemetry/sdk-node": "^0.35.1",
-        "draft-js": "^0.11.7",
+        "@next/bundle-analyzer": "^12.3.4",
```
Comment (dkarnutsch) [inline site/package.json:29]: is this related to this PR?

**Review by dkarnutsch — COMMENTED** [review]
Could we directly target the stable version is it is now released?

**Review by thomasdax98 — APPROVED** [review]
